### PR TITLE
Add support for packaging WebLayerSupport as a feature module

### PR DESCRIPTION
**Pull request by darinwf**
Changes WebLayerImpl to handle being loaded either as part of WebLayerSupport.apk or as an APK split named weblayer_support. WebLayerImpl tells the difference by checking to see what package the given appContext and removeContext belong to.

WebLayer client code is updated accordingly and importantly uses DelegateLastClassLoader (instead of PathClassLoader) to ensure that the dexPath of the split takes precedence over the dexPath of the base ClassLoader. ART gets unhappy if the parent ClassLoader does not correspond to the base APK split, so this is how we navigate that constraint.

It is critical that classes from the weblayer_support APK split get preferred by WebLayer code so that the splits can safely include duplicate symbols -- different versions of AndroidX libraries, etc.

It is also necessary for the easyprivacy list to be included as a resource of WebLayerSupport so that it can be looked up using `ContextUtils.getApplicationContext()`, so it is included in this PR. Had to switch from .proto suffix to .dat since the former is a text file format (protobuf definition), and Chromium GIT is configured to munge line endings for text files.

This change breaks support for using the system WebView in the same process that uses WebLayer.

### CODE REVIEW

**Comment by darinwf** [comment]
Still need to add the WebLayer client changes

**Comment by darinwf** [comment]
Companion PR is here: https://github.com/neevaco/neeva-android/pull/727

File: weblayer/browser/java/org/chromium/weblayer_private/WebLayerImpl.java [hunk 1]
```diff
@@ -703,16 +703,33 @@ private static Context minimalInitForContext(Context appContext, Context remoteC
...
 
+        String packageName = remoteContext.getPackageName();
+        String packagePath = remoteContext.getApplicationInfo().sourceDir;
+
+        // If the weblayer implementation is included as part of the same package as the app,
+        // then assume it is located within a split named "weblayer_support".
+        if (packageName == appContext.getPackageName()) {
+            ApplicationInfo appInfo = remoteContext.getApplicationInfo();
+            for (int i = 0; i < appInfo.splitNames.length; ++i) {
+                if (appInfo.splitNames[i].equals("weblayer_support")) {
```
Comment (yozuysal) [inline weblayer/browser/java/org/chromium/weblayer_private/WebLayerImpl.java]: Let's define these as constants up top?  (WEBLAYER_SAME_PACKAGE_SPLIT_NAME or something like that). We can move the comment there as well.

**Review by yozuysal — APPROVED** [review]

File: weblayer/browser/java/org/chromium/weblayer_private/WebLayerImpl.java (hunk 1 above)
Comment (darinwf) [reply weblayer/browser/java/org/chromium/weblayer_private/WebLayerImpl.java]: Will do.

This PR also needs some work to support x64 properly...

Comment (darinwf) [reply weblayer/browser/java/org/chromium/weblayer_private/WebLayerImpl.java]: Actually, I think we can forego x64 support for weblayer_support as a feature module and just rely on people installing `WebLayerSupport.apk` manually on x64 sims.

**Pull request merged**